src/NLog/LayoutRenderers/AppConfigLayoutRenderer.cs

```diff
@@ -0,0 +1,90 @@
+﻿// 
+// Copyright (c) 2004-2018 Jaroslaw Kowalski <jaak@jkowalski.net>, Kim Christensen, Julian Verdurmen
+// 
+// All rights reserved.
+// 
+// Redistribution and use in source and binary forms, with or without 
+// modification, are permitted provided that the following conditions 
+// are met:
+// 
+// * Redistributions of source code must retain the above copyright notice, 
+//   this list of conditions and the following disclaimer. 
+// 
+// * Redistributions in binary form must reproduce the above copyright notice,
+//   this list of conditions and the following disclaimer in the documentation
+//   and/or other materials provided with the distribution. 
+// 
+// * Neither the name of Jaroslaw Kowalski nor the names of its 
+//   contributors may be used to endorse or promote products derived from this
+//   software without specific prior written permission. 
+// 
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE 
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE 
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE 
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR 
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS 
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN 
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) 
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF 
+// THE POSSIBILITY OF SUCH DAMAGE.
+// 
+
+#if !SILVERLIGHT && !__IOS__ && !__ANDROID__ && !NETSTANDARD
+
+namespace NLog.LayoutRenderers
+{
+    using System.Collections.Specialized;
+    using System.Text;
+    using NLog.Config;
+    using NLog.Internal;
+
+    /// <summary>
+    /// Application setting.
+    /// </summary>
+    /// <remarks>
+    /// Use this layout renderer to insert the value of an application setting
+    /// stored in the application's App.config or Web.config file.
+    /// </remarks>
+    /// <code lang="NLog Layout Renderer">
+    /// ${appsetting:name=mysetting:default=mydefault} - produces "mydefault" if no appsetting
+    /// </code>
+    [LayoutRenderer("appsetting")]
+    public sealed class AppConfigLayoutRenderer : LayoutRenderer
```

Comment:
I think it would be nice if those names are in-sync ;)

Reply:
Do you mean a name clash in the layout renderer name or the class name?

Reply:
Well I changed the class-name to avoid name-clash in class-name :)

Reply:
what about a new namespace? And then in NLog 5 we could correct that.

Reply:
Will move it into Internal-then, and then it can move out with NLog 5.0

Reply:
Or AppSettingLayoutRenderer2 ;)

Reply:
Done